test/RegisterDomains.ts

```diff
@@ -0,0 +1,126 @@
+import { RegisterDomainsFixture } from "./RegisterDomains.types";
+
+const { expect } = require("chai");
+const {
+    loadFixture,
+} = require("@nomicfoundation/hardhat-toolbox/network-helpers");
+const { ethers } = require("hardhat");
+import { AddressZero } from "@ethersproject/constants";
+
+describe("RegisterDomains contract", function () {
+    let contractState: RegisterDomainsFixture;
+
+    async function deployTokenFixture(): Promise<RegisterDomainsFixture> {
+        const [owner, addr1] = await ethers.getSigners();
+        const domainsContract = await ethers.deployContract("RegisterDomains");
+        const ether = ethers.parseEther("1");
+
+        await domainsContract.waitForDeployment();
+
+        domainsContract.on("DomainRegistered", (address, domainDetails) => {
+            console.log('DomainRegistered')
+            console.log("address: ", address);
+            console.log("domainDetails: ", domainDetails);
+        });
+        domainsContract.on("DomainReleased", (address, domainDetails) => {
+            console.log(DomainReleased)
+            console.log("address: ", address);
+            console.log("domainDetails: ", domainDetails);
+        });
+
+        return { domainsContract, owner, addr1, ether };
+    }
+
+    beforeEach(async () => {
+        contractState = await loadFixture(deployTokenFixture);
+    });
+
+    describe("Deployment", function () {
+        describe('Initialization', function () {
+            it("Should set owner correctly", async function () {
+                const { domainsContract, owner } = contractState;
+
+                expect(await domainsContract.owner()).to.equal(owner.address);
+            });
+        });
+
+        describe('Domain registration', function () {
+            it("Should register domain and emit event", async function () {
+                const { domainsContract, addr1 } = contractState;
```

Comment:
because it is supposed to be used by the other clients as well, and I am demonstrating it. Since there are owner stored into the state and we should ensure we are using msg.sender and not current owner into contract

Reply:
yes, its easier, but owner is stored into state, and be sure we distinct the owner and msg.sender into the code